tools/utils/http-helpers.js

```diff
@@ -358,6 +369,47 @@ _.extend(exports, {
     } else {
       return body;
     }
-  }
+  },
 
+  getUrlWithResuming(urlOrOptions) {
+    const options = _.isObject(urlOrOptions) ? urlOrOptions : {
+      url: urlOrOptions,
+    };
+
+    const outputStream = new WritableStreamBuffer();
+
+    const MAX_ATTEMPTS = 5;
+    function attempt(triesRemaining) {
+      const headers = _.extend({}, urlOrOptions.headers, {
+        range: `bytes=${outputStream.size()}-`,
+      });
+      try {
+        return httpHelpers.request({
+          outputStream,
+          headers,
+          ...urlOrOptions,
```

Comment:
If you use `options` here, the `options.headers` property will override the `headers` property, so you might as well update `options.headers` above, and remove the `headers,` line here.